lib/syskit/network_generation/engine.rb

```diff
@@ -311,16 +311,6 @@ def find_selected_device_in_hierarchy(argument_name, leaf_task, requirements)
                 _, model, _ = leaf_task.requirements.resolved_dependency_injection.selection_for(nil, requirements)
                 if model && dev = model.arguments[argument_name]
                     return dev
-                else
-                    devices = Set.new
```

Comment:
Just to clarify my position: I am not against disabling auto-selection of anything BUT it should IMO be optional. I start to use syskit to replace small ruby scripts, and auto-selection there is a blessing